streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -17,73 +17,72 @@
 
 package org.apache.kafka.streams.kstream;
 
+import org.apache.kafka.common.annotation.InterfaceStability;
 import org.apache.kafka.common.serialization.Serde;
 import org.apache.kafka.streams.KeyValue;
 
 /**
- * KTable is an abstraction of a change log stream from a primary-keyed table.
+ * KTable is an abstraction of a <i>changelog stream</i> from a primary-keyed table.
  *
  * @param <K> Type of primary keys
  * @param <V> Type of value changes
  */
+@InterfaceStability.Unstable
 public interface KTable<K, V> {
 
     /**
-     * Creates a new instance of KTable consists of all elements of this stream which satisfy a predicate
+     * Create a new instance of {@link KTable} consists of all elements of this stream which satisfy a predicate.
      *
-     * @param predicate the instance of Predicate
-     * @return the instance of KTable with only those elements that satisfy the predicate
+     * @param predicate     the instance of {@link Predicate}
      */
     KTable<K, V> filter(Predicate<K, V> predicate);
 
     /**
-     * Creates a new instance of KTable consists all elements of this stream which do not satisfy a predicate
+     * Create a new instance of {@link KTable} consists all elements of this stream which do not satisfy a predicate
```

Comment:
Also, I'd prefer if we'd rename `filterOut` to `filterNot` because users familiar to e.g. Scala will see that the functionality is equivalent (i.e. why invent the name `filterOut` if `filterNot` is already established elsewhere).

WDYT, @ijuma ?